x-pack/solutions/security/plugins/security_solution/public/entity_analytics/pages/entity_analytics_management_page.tsx

```diff
@@ -47,12 +47,20 @@ export const EntityAnalyticsManagementPage = () => {
   const [isLoading, setIsLoading] = useState(false);
   const { mutate: scheduleNowRiskEngine } = useScheduleNowRiskEngineMutation();
   const { addSuccess, addError } = useAppToasts();
+  const userHasRequiredPrivileges =
+    'hasAllRequiredPrivileges' in privileges && privileges.hasAllRequiredPrivileges;
+  const btnIsDisabled = !currentRiskEngineStatus || isLoading || !userHasRequiredPrivileges;
 
   const handleRunEngineClick = async () => {
+    if (!userHasRequiredPrivileges) {
+      addError(i18n.CHECK_PRIVILEGES, {
+        title: i18n.CHECK_PRIVILEGES,
+      });
+      return;
+    }
```

Comment:
Why do we need the extra check if the button is already disabled?
The error msg `Check privileges` is vague. If we need this code, we should create a more descriptive message.